receiver/receiverhelper/scrapercontroller.go

```diff
@@ -25,96 +26,138 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 )
 
-// ScraperControllerOption apply changes to internal options.
-type ScraperControllerOption func(*scraperController)
+// ScraperControllerConfig is the configuration of a scraper controller.
+// Specific scrapers must implement this interface and will typically embed
+// ScraperControllerSettings struct or a struct that extends it.
+type ScraperControllerConfig interface {
+	CollectionInterval() time.Duration
```

Comment:
In order for this to work, you need also to embed the `configmodels.Receiver` here.

Not sure what is the benefit of having this interface, can we simply remove it?

Reply:
Ah yes good point. I'll update this to embed `configmodels.Receiver`.

I do think we want to keep this interface as it's much easier for users to embed `ScraperControllerSettings` in their config and call `NewScraperControllerReceiver` if we have it.

With interface:
```go
NewScraperControllerReceiver(cfg ScraperControllerConfig, ...

# calling
NewScraperControllerReceiver(cfg, ...
```

Without interface:
```go
NewScraperControllerReceiver(cfg configmodels.Receiver, scs ScraperControllerSettings, ...

# calling
NewScraperControllerReceiver(cfg, cfg.ScraperControllerSettings, ...
```
OR
```go
NewScraperControllerReceiver(cfg configmodels.Receiver, collectionInterval time.Duration, ...

# calling
NewScraperControllerReceiver(cfg, cfg.ScraperControllerSettings.CollectionInterval, ...
```
OR
```go
NewScraperControllerReceiver(cfg configmodels.Receiver, ...

# calling
NewScraperControllerReceiver(cfg, ..., WithCollectionInterval(cfg.ScraperControllerSettings.CollectionInterval))
```

Reply:
What about passing directly the ScraperControllerSettings ?